fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergExternalTable.java

```diff
@@ -63,10 +68,14 @@ public class IcebergExternalTable extends ExternalTable implements MTMVRelatedTa
     private Table table;
     private boolean isValidRelatedTableCached = false;
     private boolean isValidRelatedTable = false;
+    private boolean isTable;
+    private boolean isView;
 
     public IcebergExternalTable(long id, String name, String remoteName, IcebergExternalCatalog catalog,
             IcebergExternalDatabase db) {
         super(id, name, remoteName, catalog, db, TableType.ICEBERG_EXTERNAL_TABLE);
+        isTable = catalog.tableExist(null, getRemoteDbName(), getRemoteName());
```

Comment:
I think we should move these 2 lines to `makeSureInitialized()`
And in `isTable()` and `isView()` method, call `makeSureInitialized()` first